src/Forum/ForumServiceProvider.php

```diff
@@ -204,7 +204,7 @@ protected function setDefaultRoute(RouteCollection $routes)
         $factory = $this->container->make(RouteHandlerFactory::class);
         $defaultRoute = $this->container->make('flarum.settings')->get('default_route');
 
-        if (isset($routes->getRoutes()['GET'][$defaultRoute]['handler'])) {
+        if (isset($routes->getRouteData()[0]['GET'][$defaultRoute]['handler'])) {
             $toDefaultController = $routes->getRoutes()['GET'][$defaultRoute]['handler'];
```

Comment:
$toDefaultController = $routes->getRoutes()['GET'][$defaultRoute]['handler'];
$toDefaultController = $routes->getRouteData()[0]['GET'][$defaultRoute]['handler'];
Does it need to be modified ？

Reply:
@SychO9 This is one for you, Mr PHP wizard

Reply:
Hello @xiaowu771086986,

Yea we hadn't noticed it at the time and tests didn't catch it, but we did actually fix it not long after here: https://github.com/flarum/core/commit/1cca4e8564194d3941ab76321976ca55ac58ce19

Is this what you meant ?

Reply:
yes ,thank you !